modules/light-clients/08-wasm/types/update_test.go

```diff
@@ -652,115 +661,99 @@ func (suite *TypesTestSuite) TestUpdateStateGrandpa() {
 // 	}
 // }
 
-/* func (suite *TypesTestSuite) TestUpdateStateOnMisbehaviourGrandpa() {
-	var (
-		ok          bool
-		clientMsg   exported.ClientMessage
-		clientState exported.ClientState
-	)
+func (suite *TypesTestSuite) TestUpdateStateOnMisbehaviour() {
+	var clientMsg exported.ClientMessage
 
 	testCases := []struct {
-		name    string
-		setup   func()
-		expPass bool
+		name               string
+		malleate           func()
+		panicErr           error
+		updatedClientState []byte
 	}{
 		{
-			"successful update",
+			"success: no update",
 			func() {
-				data, err := base64.StdEncoding.DecodeString(suite.testData["misbehaviour"])
-				suite.Require().NoError(err)
-				clientMsg = &types.ClientMessage{
-					Data: data,
-				}
+				suite.mockVM.RegisterSudoCallback(types.UpdateStateOnMisbehaviourMsg{}, func(_ wasmvm.Checksum, _ wasmvmtypes.Env, _ []byte, store wasmvm.KVStore, _ wasmvm.GoAPI, _ wasmvm.Querier, _ wasmvm.GasMeter, _ uint64, _ wasmvmtypes.UFraction) (*wasmvmtypes.Response, uint64, error) {
```

Comment:
we should unmarshal `sudoMsg []byte` as the the `UpdateStateOnMisbehaviourMsg` and ensure its values are as we expect our go layer to encode it

Reply:
needs to go into a `SudoMsg` wrapper first I guess, and then check that `UpdateStateOnMisbehaviourMsg` is not nil..etc